packages/venia-ui/lib/components/OrderHistoryPage/collapsedImageGallery.js

```diff
@@ -0,0 +1,49 @@
+import React, { useMemo } from 'react';
+import { useCollapsedImageGallery } from '@magento/peregrine/lib/talons/OrderHistoryPage/useCollapsedImageGallery';
+
+import { mergeClasses } from '../../classify';
+import defaultClasses from './collapsedImageGallery.css';
+import imageGalleryOperations from './collapsedImageGallery.gql';
+import Image from '../Image';
+
+const CollapsedImageGallery = props => {
+    const { items } = props;
+
+    const talonProps = useCollapsedImageGallery({
+        items,
+        ...imageGalleryOperations
+    });
+    const { imageData } = talonProps;
+
+    const classes = mergeClasses(defaultClasses, props.classes);
+    const remainderCount = items.length - 4;
+
+    const imageElements = useMemo(() => {
+        if (imageData) {
+            const { products } = imageData;
+            const { items } = products;
+
+            const baseImageElements = items.map(item => {
```

Comment:
Shouldn't this be `items.slice(0, 4).map` instead? You don't wanna map over all the items right? Because, after 4, you just want to show the count.

Reply:
With the constant added in a14f640, you can see I'm cleverly only fetching the first four.